include/materials/ConcreteASREigenstrain.h

```diff
@@ -0,0 +1,109 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef CONCRETEASREIGENSTRAIN_H
+#define CONCRETEASREIGENSTRAIN_H
+
+#include "ConcreteExpansionEigenstrainBase.h"
+#include "RankTwoTensor.h"
+
+/**
+ * Computes the volumetric expansion eigenstrain due to alkali-silica reaction.
+ */
+
+class ConcreteASREigenstrain : public ConcreteExpansionEigenstrainBase
+{
+public:
+  ConcreteASREigenstrain(const InputParameters & parameters);
+
+  void initQpStatefulProperties() override;
+  bool needStressEigenvalues() override { return true; }
+  Real computeQpVolumetricStrain() override;
+
+  /**
+   * Compute the residual for the computation of the ASR reaction extent.
+   * @param qp Current quadrature point id
+   * @param scalar Current iterative value of the ASR reaction extent
+   * @return residual, which is the error in the new ASR reaction extent
```

Comment:
To a non concrete person, extent isn't as clear as it could be. Do you mean how far along the reaction is?

Reply:
Yes.  I added an expanded comment to the _ASR_extent variable to explain. It's also well-documented in the md file for this class.